streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -776,9 +776,10 @@ void runOnce() {
 
                 log.debug("{} punctuators ran.", punctuated);
 
+                final long beforeCommitTs = now;
                 final int committed = maybeCommit();
                 totalCommittedSinceLastSummary += committed;
-                final long commitLatency = advanceNowAndComputeLatency();
+                final long commitLatency = Math.max(now - beforeCommitTs, 0);
```

Comment:
Also kind of a nit, since technically this does work, but wouldn't it make more sense to just remove the `advanceNowAndComputeLatency` call in `maybeCommit`, and then just call `advancedNowAndComputeLatency` here as before? Otherwise we're just computing the latency inside `maybeCommit` for no reason, and throwing out the result.

Reply:
Well, we need to advance time _within_ `maybeCommit()` to make sure we compute `lastCommitMs` correctly (what was exactly the fix that introduced the commit-latency metric regression).

Reply:
We could replace `advancedNowAndComputeLatency` by `now = time.milliseconds()` in `maybeCommit()` though.

Reply:
But currently we call `advanceNowAndComputeLatency` pretty much at the end of the `maybeCommit` method, so there shouldn't be a noticeable difference between advancing `now` at the end of `maybeCommit` vs advancing it immediately after `maybeCommit` returns, right?
I'm also ok with just advancing `now` inside `maybeCommit`. The main thing that felt off was just that we compute the latency for no reason inside `maybeCommit`, and ignore the result

Reply:
(and similarly move the `lastCommitMs = now` to after `maybeCommit` returns)

Reply:
> so there shouldn't be a noticeable difference between advancing now at the end of maybeCommit vs advancing it immediately after maybeCommit returns, right?

Well, not from a runtime point of view.

> (and similarly move the lastCommitMs = now to after maybeCommit returns)

Exactly: we need to first advance `now` and than update `lastCommitMs` -- but I would rather not update `lastCommitMs` outside of `maybeCommit`, because it breaks encapsulation?

> I'm also ok with just advancing now inside maybeCommit. The main thing that felt off was just that we compute the latency for no reason inside maybeCommit, and ignore the result

Yes, that is a little odd -- as I said, happy to change `advancedNowAndComputeLatency` to `now = time.milliseconds()` within `maybeCommit()`.

Reply:
> Yes, that is a little odd -- as I said, happy to change advancedNowAndComputeLatency to now = time.milliseconds() within maybeCommit().

Sounds good to me, let's do it